sycl/include/sycl/ext/oneapi/experimental/device_architecture.hpp

```diff
@@ -8,27 +8,27 @@ namespace ext::oneapi::experimental {
 
 enum class architecture {
   x86_64,
-  intel_gpu_bdw,
-  intel_gpu_skl,
-  intel_gpu_kbl,
-  intel_gpu_cfl,
-  intel_gpu_apl,
-  intel_gpu_glk,
-  intel_gpu_whl,
-  intel_gpu_aml,
-  intel_gpu_cml,
-  intel_gpu_icllp,
-  intel_gpu_tgllp,
-  intel_gpu_rkl,
-  intel_gpu_adl_s,
-  intel_gpu_rpl_s,
-  intel_gpu_adl_p,
-  intel_gpu_adl_n,
-  intel_gpu_dg1,
-  intel_gpu_acm_g10,
-  intel_gpu_acm_g11,
-  intel_gpu_acm_g12,
-  intel_gpu_pvc,
+  intel_gpu_bdw = 0x02000000,
```

Comment:
@gmlueck, theoretical problem which can be raised somewhere in the future: because we have mapping of architectures to some architecture IDs to support our host APIs, we may face the situation when we have the same architecture IDs for some architectures of different vendors, e.g., two AMD and NVIDIA architectures may have the same IDs, and there will be an incorrect results for them. Is it a possible future problem which may be addressed in the specification?

Reply:
I was not expecting to change the enum values like this.  Instead, I was thinking that the PI plugins would have a translation table such as:

```
// In Level Zero: All devices are Intel GPU, so no need to check device type
std::map<uint32_t, syc;::ext::oneapi::experimental::architecture> MapDeviceIpToArch = {/*...*/};

uint32_t ip = Device->ZeDeviceIpVersionExt->ipVersion;
return MapDeviceIpToArch[ip];
```

```
// In OpenCL: Must check device type first (GPU vs. CPU):
std::map<uint32_t, syc;::ext::oneapi::experimental::architecture> MapGpuDeviceIpToArch = {/*...*/};

if (/*device type GPU*/) {
  uint32_t ip = /*...*/;
  return MapGpuDeviceIpToArch[ip];
}
else /*device type is CPU*/ {
  // In future, we may provide different enums for CPU architectures.
  return sycl::ext::oneapi:experimental::architecture::x86_64;
}
```

By doing it this way, there is no problem with overlap between architecture IDs across different device families.

BTW, there is no update to the OpenCL PI plugin in this PR.  Shouldn't we add that?

Reply:
I will implement like this for each backend. Another advantage of this is that not every vendor may return IP version as `uint32_t`, so it's good to have maps like this.

> there is no update to the OpenCL PI plugin in this PR. Shouldn't we add that?

For OpenCL, https://github.com/intel/llvm/blob/811f8b3a78af363fe5f38d35e25b7379fae563d1/sycl/include/sycl/detail/pi.h#L359 this line is enough. We just call `clGetDeviceInfo` underneath with this parameter, and it's value, `0x4250` is in accordance with [cl_intel_device_attribute_query](https://registry.khronos.org/OpenCL/extensions/intel/cl_intel_device_attribute_query.html) spec

Reply:
Implemented